http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/ResponseHandler.java

```diff
@@ -246,11 +245,16 @@ public void onNext(HttpContent httpContent) {
                     if (isDone.get()) {
                         return;
                     }
-                    // Needed to prevent use-after-free bug if the subscriber's onNext is asynchronous
-                    ByteBuffer b = copyToByteBuffer(httpContent.content());
-                    httpContent.release();
-                    subscriber.onNext(b);
-                    channelContext.read();
+
+                    try {
+                        // Needed to prevent use-after-free bug if the subscriber's onNext is asynchronous
+                        ByteBuffer b = copyToByteBuffer(httpContent.content());
+                        httpContent.release();
+                        subscriber.onNext(b);
```

Comment:
I think we might need to move this out of the try-catch. According to the spec, if `onNext` throws, we need to treat the subscription as cancelled, and we can't call `onError` in that case (see 2.13 https://github.com/reactive-streams/reactive-streams-jvm#2-subscriber-code)

Reply:
Yeah, fixed